library/logrotate.py

```diff
@@ -160,11 +160,32 @@ def validate_config(module):
     fh.close()
 
     LOGROTATE = module.get_bin_path('logrotate', True)
-    rc, out, err = module.run_command('%s -d %s' % (LOGROTATE, temppath), check_rc=True)
+
+    # read not only the file to validate but the default configuration because
+    # some defaults are needed to validate, notably `su` directive
+    default_config_path = get_default_config_path(module)
+    rc, out, err = module.run_command('%s -d %s %s' % (LOGROTATE, default_config_path, temppath), check_rc=True)
     os.unlink(temppath)
     if rc != 0:
         module.fail_json(msg='failed to validate config for: %s' % (name), stdout=out, stderr=err)
 
+def get_default_config_path(module):
+    """Look for the default configuration and return the first one found"""
+    locations = [
+        # Linux
+        '/etc/logrotate.conf',
+        # FreeBSD
+        '/usr/local/etc/logrotate.conf'
+        ]
+    found = ''
+    for path in locations:
```

Comment:
no, it does not. i was looking for get the value of `ansible_os_family` within a module but failed. the logic is just guessing OS by the location.